spec/support/matchers/accessibility.rb

```diff
@@ -24,6 +24,50 @@
   end
 end
 
+RSpec::Matchers.define :have_valid_markup do
+  def page_html
+    if page.driver.is_a?(Capybara::Selenium::Driver)
+      # Here be dragons:
+      # 1. ChromeDriver repairs most invalid HTML, thus defeating the purpose of this test
+      # 2. Switching drivers does not preserve the original session, hence the cookie management
+      # 3. The domain name differs between Rack and ChromeDriver testing environments
+      # 4. Methods stubbed for JS tests carry over when switching drivers (see `rails_helper.rb`)
+      cookies = page.driver.browser.send(:bridge).cookies
+      session_value = cookies.find { |c| c['name'] == '_identity_idp_session' }&.[]('value')
+      original_path_with_params = URI.parse(current_url).request_uri
+      Capybara.using_driver(:desktop_rack_test) do
+        domain_name = IdentityConfig.store.domain_name
+        default_url_options = Rails.application.routes.default_url_options
+        allow(IdentityConfig.store).to receive(:domain_name).and_call_original
+        allow(Rails.application.routes).to receive(:default_url_options).and_call_original
+        page.driver.browser.set_cookie "_identity_idp_session=#{session_value}" if session_value
+        page.driver.get(original_path_with_params)
+        allow(IdentityConfig.store).to receive(:domain_name).and_return(domain_name)
+        allow(Rails.application.routes).to receive(:default_url_options).
+          and_return(default_url_options)
+        page.html
+      end
+    else
+      page.html
+    end
+  end
+
+  def page_markup_syntax_errors
+    return @page_markup_syntax_errors if defined?(@page_markup_syntax_errors)
+    @page_markup_syntax_errors = Nokogiri::HTML5(page_html, max_errors: 20).errors
+  end
+
+  match { |_page| page_markup_syntax_errors.blank? }
+
+  failure_message do |page|
+    <<-STR.strip_heredoc
```

Comment:
same as #7642, I think we can simplify
```suggestion
    <<~STR
```

Reply:
> same as #7642, I think we can simplify

Updated in rebased 260726ad8.